modules/risks-setting-higher-process-id-limits.adoc

```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+//
+// * rosa_cluster_admin/rosa-configuring-pid-limits.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="risks-setting-higher-process-id-limits_{context}"]
+= Risks of setting higher process ID limits for {product-title} pods
+
+The `podPidsLimit` parameter for a pod controls the maximum number of processes and threads that can run simultaneously in that pod.
+
+You can increase the value for `podPidsLimit` from the default of 4,096 to a maximum of 16,384. Changing this value may incur downtime for applications, because changing the `podPidsLimit` requires rebooting the affected node.
+
+If you are running a large number of pods per node, and you have a high `podPidsLimit` value on your nodes, you risk exceeding the PID maximum for the node.
+
+To find the maximum number of pods that you can run simultaneously on a single node without exceeding the PID maximum for the node, divide 3,650,000 by your `podPidsLimit` value. For example, if your `podPidsLimit` value is 1,048,576, and you expect the pods to use close to that number of process IDs, you can safely run three pods on a single node.
```

Comment:
I love the idea of giving an example. Can we adjust the example to align with maximum supported value of 16,384? Introducing a number 1,048,576 may confuse or wrongly give an impression that podPidsLimit support this value; we've just above mentioned that the maximum generally allowed is 16,384. Perhaps we use a number within the allowed range?

Reply:
@robpblake Does this example make sense? What does the 1,048,576 number represent here? How could it be this high if we only support a max of 16,384?

Reply:
@bhardesty I'm fairly sure there is no specific meaning to the `1,048,576` number. I think it was just an example to show how to work out roughly how many pods could run on a single node given the `podPidsLimit` you apply.

I agree that we should use an example with a `podPidsLimit` <= 16,384 as that the upper bound by default.

> How could it be this high if we only support a max of 16,384?

For certain customers, we have the ability to assign a capability to their organization (`capability.organization.bypass_pids_limits`) that allows them to exceed the default 16,384 limit (up to about 3.5. million). This was required as certain important customers already have workloads that go far beyond the 16,384 requirement and asking them to re-architect their workloads to use this feature is not feasible.

So in summary:

- Most customers will stick with the default range and be encouraged to correctly architect their applications to work with this
- For some customers, we'll let them exceed the default limit, particularly if there are legacy workloads that cannot be easily migrated.

cheers,

Rob

Reply:
I updated the example to use the max PID limit of 16,384.